networkx/algorithms/community/nodal_roles_nx.py

```diff
@@ -0,0 +1,128 @@
+from networkx.algorithms.community import louvain
+import networkx as nx
+
+def convert_dictionary(dictionary):
+	'''
+	Converts dictionary with nodes as keys to modules as keys
+
+	------
+	Inputs
+	------
+	dictionary =  dictionary output from Louvain community detection
+
+	------
+	Output
+	------
+	Dictionary with modules as keys, nodes as values
+
+	'''
+
+	new_dict = {}
+	for m,n in zip(dictionary.values(),dictionary.keys()):
+	    try:
+	        new_dict[m].append(n)
+	    except KeyError:
+	        new_dict[m] = [n]
+	return new_dict
```

Comment:
Simpler: `return {k : list(v) for k, v in groupby(sorted(d, key=d.get), key=d.get)}`.